File: internal/model/multimediametadata.go
package model

import "time"

type MultimediaMetadata struct {
	Id          string    `json:"id"`
	Name        string    `json:"name"`
	Type        string    `json:"type"`
	SizeInKB    float32   `json:"sizeInKB"`
	CreatedAt   time.Time `json:"createdAt"`
	LastChanged time.Time `json:"lastChanged"`
	Username    string    `json:"username"`
}

func NewMultimediaMetadata(id string, name string, Type string, sizeInKB float32,
	createdAt time.Time, lastChanged time.Time, username string) *MultimediaMetadata {
	return &MultimediaMetadata{
		Id:          id,
		Name:        name,
		Type:        Type,
		SizeInKB:    sizeInKB,
		CreatedAt:   createdAt,
		LastChanged: lastChanged,
		Username:    username}
}

File: internal/model/multimedia_metadata.py
import json


class MultimediaMetadata:
    def __init__(self, id, name, type, size_in_kb, created_at, last_changed, username, description):
        self.id = id
        self.name = name
        self.type = type
        self.size_in_kb = size_in_kb
        self.created_at = created_at
        self.last_changed = last_changed
        self.username = username
        self.description = description

    def to_json(self):
        return json.dumps(self.__dict__)

    @staticmethod
    def from_json(json_str):
        data = json.loads(json_str)
        return MultimediaMetadata(**data)

    def to_dict(self):
        return {
            "id": self.id,
            "name": self.name,
            "type": self.type,
            "size_in_kb": self.size_in_kb,
            "created_at": self.created_at,
            "last_changed": self.last_changed,
            "username": self.username,
            "description": self.description
        }


class MultimediaDisplay(MultimediaMetadata):
    def __init__(self, id, name, type, size_in_kb, created_at, last_changed, username, description, data_url=""):
        super().__init__(id, name, type, size_in_kb, created_at, last_changed, username, description)
        self.data_url = data_url

    def to_dict(self):
        multimedia_output = super().to_dict()
        multimedia_output["data_url"] = self.data_url
        return multimedia_output